transport-native-epoll/src/main/c/netty_unix_socket.c

```diff
@@ -708,6 +730,12 @@ static JNINativeMethod* createDynamicMethodsTable(const char* packagePrefix) {
     dynamicMethod->signature = netty_unix_util_prepend("(IJII)L", dynamicTypeName);
     dynamicMethod->fnPtr = (void *) netty_unix_socket_recvFromAddress;
     free(dynamicTypeName);
+    ++dynamicMethod;
+    dynamicTypeName = netty_unix_util_prepend(packagePrefix, "io/netty/channel/unix/PeerCredentials;");
+    dynamicMethod->name = "peerCredentials";
+    dynamicMethod->signature = netty_unix_util_prepend("(ILio/netty/channel/unix/PeerCredentials;)I", "");
```

Comment:
Ok think I now understand the intent of this code.  The goal is to provide an override to the netty class for use with shading.  So by me passing the netty object (PeerCredentials) in as a parameter rather than returning it makes this a bit more complicated.  

So to address this and your last comment I will change the native signature to return the PeerCredentials object as suggested below.